src/nbl/ext/ImGui/ImGui.cpp

```diff
@@ -570,82 +603,265 @@ namespace nbl::ext::imgui
+							auto& element = requestData.elements.emplace_back();
+
+							indirect.firstIndex = pcmd->IdxOffset + globalIOffset;
+							indirect.firstInstance = drawID; // use base instance as draw ID
+							indirect.indexCount = pcmd->ElemCount;
+							indirect.instanceCount = 1u;
+							indirect.vertexOffset = pcmd->VtxOffset + globalVOffset;
+
+							const auto scissor = clip.getScissor(clipRectangle);
+
+							auto packSnorm16 = [](float ndc) -> int16_t
+							{
+								return std::round<int16_t>(std::clamp(ndc, -1.0f, 1.0f) * 32767.0f); // TODO: ok encodePixels<asset::EF_R16_SNORM, double>(void* _pix, const double* _input) but iirc we have issues with our encode/decode utils
+							};
+
+							const auto vMin = trs.toNDC(core::vector2df_SIMD(scissor.offset.x, scissor.offset.y));
+							const auto vMax = trs.toNDC(core::vector2df_SIMD(scissor.offset.x + scissor.extent.width, scissor.offset.y + scissor.extent.height));
+
+							element.aabbMin.x = packSnorm16(vMin.x);
+							element.aabbMin.y = packSnorm16(vMin.y);
+							element.aabbMax.x = packSnorm16(vMax.x);
+							element.aabbMax.y = packSnorm16(vMax.y);
+							// element.texId = 0; // use defaults from constructor
+
+							++drawID;
+						}
+					}
+
+					globalIOffset += cmd_list->IdxBuffer.Size;
+					globalVOffset += cmd_list->VtxBuffer.Size;
+				}
+
+				requestData.iBufferSize = drawData->TotalIdxCount * sizeof(ImDrawIdx);
+				requestData.vBufferSize = drawData->TotalVtxCount * sizeof(ImDrawVert);
+				requestData.vPadding = requestData.getVerticesOffset() % 4u;
+				assert(requestData.validate());
+			}
+			
+			IGPUBuffer::SCreationParams mdiCreationParams = {};
+			mdiCreationParams.usage = nbl::core::bitflag(nbl::asset::IBuffer::EUF_INDIRECT_BUFFER_BIT) | nbl::asset::IBuffer::EUF_INDEX_BUFFER_BIT | nbl::asset::IBuffer::EUF_VERTEX_BUFFER_BIT | nbl::asset::IBuffer::EUF_SHADER_DEVICE_ADDRESS_BIT  | nbl::asset::IBuffer::EUF_INLINE_UPDATE_VIA_CMDBUF;
+			mdiCreationParams.size = requestData.getMDIBufferSize();
+
+			auto mdiBuffer = m_mdiBuffers[frameIndex];
+			const uint32_t drawCount = requestData.elements.size();
+			const auto mdicBSize = drawCount * sizeof(VkDrawIndexedIndirectCommand);
+			const auto elementsBSize = drawCount * sizeof(PerObjectData);
+			const uint64_t elementsBOffset = requestData.getElementsOffset();
+
+			// create or resize the mdi buffer
+			if (!static_cast<bool>(mdiBuffer) || mdiBuffer->getSize() < mdiCreationParams.size)
+			{
+				mdiBuffer = m_device->createBuffer(std::move(mdiCreationParams));
+
+				auto mReqs = mdiBuffer->getMemoryReqs();
+				mReqs.memoryTypeBits &= m_device->getPhysicalDevice()->getUpStreamingMemoryTypeBits();
+				
+				auto memOffset = m_device->allocate(mReqs, mdiBuffer.get(), core::bitflag<IDeviceMemoryAllocation::E_MEMORY_ALLOCATE_FLAGS>(IDeviceMemoryAllocation::E_MEMORY_ALLOCATE_FLAGS::EMAF_DEVICE_ADDRESS_BIT));
 				assert(memOffset.isValid());
 			}
```

Comment:
a buffer of a fixed size should be created up front.

@Erfan-Ahmadi will teach you how to handle overflows (when you have so much data that it won't fit the memory available in the buffer)